src/components/post/PostHighlights.vue

```diff
@@ -0,0 +1,240 @@
+<template>
+  <div class="modal" :class="{ 'is-active': isActive }">
+    <div class="modal-content">
+      <div class="modal-header">
+        <strong class="title">{{ commentCount || 0 }}  {{commentCount == 1 ? 'highlight' : 'highlights'}}</strong>
+        <div class="cursor-pointer" @click="onClose">
+          <i class="fa fa-lg fa-close"></i>
+        </div>
+      </div>
+      <div ref="modal-body" class="modal-body">
+
+        <comment-highlight
+          v-if="canComment"
+          :forumThreadId="threadId"
+          :highlight="highlight"
+          :autoFocus="true" />
+
+        <comments-list
+          :filter="'highlight'"
+          :post="post"
+          :forumThreadId="threadId"
+          :commentCount="commentCount"
+          :comments="commentsSorted" />
+
+      </div>
+      <div class="modal-footer">
+        <button class="btn-close" @click="onClose">
+          Close
+        </button>
+      </div>
+    </div>
+    <div class="modal-overlay" @click="onClose" />
+  </div>
+</template>
+
+<script>
+import clone from 'lodash/clone'
+import isEqual from 'lodash/isEqual'
+import CommentHighlight from '@/components/comment/CommentHighlight'
+import CommentsList from '@/components/comment/CommentsList'
+
+export default {
+  props: {
+    post: {
+      type: Object
+    },
+    comments: {
+      type: Array,
+      default: () => [],
+    },
+    commentCount: {
+      type: Number,
+      default: 0,
+    },
+    isDone: {
+      type: Boolean,
+      default: false,
+    },
+    highlight: {
+      type: String,
+      defaults: '',
+    }
+  },
+
+  components: {
+    CommentsList,
+    CommentHighlight,
+  },
+
+  data () {
+    return {
+      isActive: false,
+      commentId: null,
+      threadId: null,
+    }
+  },
+
+  watch: {
+    $route (to, from) {
```

Comment:
It looks like there's another option for this `watch` case:
https://router.vuejs.org/guide/advanced/navigation-guards.html#in-component-guards
I don't know if it works in our app

Reply:
Does this option have a benefit over how it's currently implemented?